test/instaskip/impl/from_eskip_facts.clj

```diff
@@ -6,75 +6,99 @@
+               :predicates [{:name "pred1" :args []}
+                            {:name "pred2" :args [{:value "Hello" :type "string"}]}]
+               :endpoint   ""}]))
 
-(fact "eskip->json parses a simple route"
-  (eskip->json "hello1: pred1() -> <shunt>;") =>
-    (str "[{\"name\":\"hello1\","
-         "\"predicates\":[{\"name\":\"pred1\",\"args\":[]}],"
-         "\"filters\":[],"
-         "\"endpoint\":\"\"}]"))
+(facts "eskip->json"
+       (fact "parses a match all route"
+             (eskip->json "hello1: * -> <shunt>;") =>
+             (str "[{\"name\":\"hello1\","
+                  "\"predicates\":[{\"name\":\"*\",\"args\":[]}],"
+                  "\"filters\":[],"
+                  "\"endpoint\":\"\"}]"))
 
-(fact "eskip->json parses a simple route wiht an endpoint"
-      (eskip->json "hello1: pred1() -> \"http://www.hello.com/\";") =>
-      (str "[{\"name\":\"hello1\","
-           "\"predicates\":[{\"name\":\"pred1\",\"args\":[]}],"
-           "\"filters\":[],"
-           "\"endpoint\":\"http:\\/\\/www.hello.com\\/\"}]"))
+       (fact "parses a simple route"
+             (eskip->json "hello1: pred1() -> <shunt>;") =>
+             (str "[{\"name\":\"hello1\","
+                  "\"predicates\":[{\"name\":\"pred1\",\"args\":[]}],"
+                  "\"filters\":[],"
+                  "\"endpoint\":\"\"}]"))
 
-(fact "eskip->json parses a route with a predicate"
-  (eskip->json "hello1: pred1(\"hello\", 3, 4.2, 0, /^.*$/) -> <shunt>;") =>
-    (str "[{\"name\":\"hello1\","
-         "\"predicates\":[{\"name\":\"pred1\",\"args\":[\"hello\",3,4.2,0,\"\\/^.*$\\/\"]}],"
-         "\"filters\":[],"
-         "\"endpoint\":\"\"}]"))
+       (fact "parses a simple route wiht an endpoint"
+             (eskip->json "hello1: pred1() -> \"http://www.hello.com/\";") =>
+             (str "[{\"name\":\"hello1\","
+                  "\"predicates\":[{\"name\":\"pred1\",\"args\":[]}],"
+                  "\"filters\":[],"
+                  "\"endpoint\":\"http:\\/\\/www.hello.com\\/\"}]"))
 
-(fact "eskip->json parses a route with many predicates"
-  (eskip->json "hello1: pred1(\"hello\") && pred2(2) -> <shunt>;") =>
-      (str "[{\"name\":\"hello1\","
-           "\"predicates\":[{\"name\":\"pred1\",\"args\":[\"hello\"]},{\"name\":\"pred2\",\"args\":[2]}],"
-           "\"filters\":[],"
-           "\"endpoint\":\"\"}]"))
+       (fact "parses a route with a predicate"
+             (eskip->json "hello1: pred1(\"hello\", 3, 4.2, 0, /^.*$/) -> <shunt>;") =>
+             (str "[{\"name\":\"hello1\","
+                  "\"predicates\":[{\"name\":\"pred1\","
+                  "\"args\":[{\"value\":\"hello\",\"type\":\"string\"},{\"value\":\"3\",\"type\":\"number\"},{\"value\":\"4.2\",\"type\":\"number\"},{\"value\":\"0\",\"type\":\"number\"},{\"value\":\"\\/^.*$\\/\",\"type\":\"regex\"}]}],"
+                  "\"filters\":[],"
+                  "\"endpoint\":\"\"}]"))
```

Comment:
Those tests are nasty. I think it would be better and more stable to convert the json to clojure data structures (edn) and do a comparison on that level. It removes you one step from the output of `eskip->json`, but it makes the tests a lot more stable and easier to refactor.